test/occupation.jl

```diff
@@ -1,11 +1,24 @@
 using Test
-using DFTK: smearing_functions, find_fermi_level, Model, PlaneWaveBasis
-using DFTK: find_occupation, find_occupation_bandgap
-using DFTK: smearing_fermi_dirac, load_psp, Species, bzmesh_ir_wedge
-using DFTK: smearing_methfessel_paxton_1
+using DFTK
+using SpecialFunctions
 
 include("testcases.jl")
 
+@testset "Smearing functions" begin
+    for m in DFTK.Smearing.smearing_methods
+        @test DFTK.Smearing.occupation(m(), -Inf) == 1
+        @test DFTK.Smearing.occupation(m(), Inf) == 0
+        x = .04
+        ε = 1e-8
+        @test abs((DFTK.Smearing.occupation(m(), x+ε) - DFTK.Smearing.occupation(m(), x))/ε -
+                  DFTK.Smearing.occupation_derivative(m(), x)) < 1e-4
+
+        # entropy functions should satisfy s' = x f'
```

Comment:
What helped me to understand this was https://www.vasp.at/vasp-workshop/k-points.pdf page 12. Could be added here.